src/ray/gcs/gcs_server/gcs_monitor_server.h

```diff
@@ -34,8 +36,17 @@ class GcsMonitorServer : public rpc::MonitorServiceHandler {
                               rpc::DrainAndKillNodeReply *reply,
                               rpc::SendReplyCallback send_reply_callback) override;
 
+  void HandleGetSchedulingStatus(rpc::GetSchedulingStatusRequest request,
+                                 rpc::GetSchedulingStatusReply *reply,
+                                 rpc::SendReplyCallback send_reply_callback) override;
+
  private:
+  void PopulateNodeStatuses(rpc::GetSchedulingStatusReply *reply) const;
+  void PopulateResourceDemands(rpc::GetSchedulingStatusReply *reply) const;
+
   std::shared_ptr<GcsNodeManager> gcs_node_manager_;
+  ClusterResourceManager &cluster_resource_manager_;
+  std::shared_ptr<GcsResourceManager> gcs_resource_manager_;
```

Comment:
nit: this is a little bit strange. I mean why two are shared ptr and one by reference. but not big issue.

Reply:
IMO the reference makes sense, the real question to me is why are 2 of these shared ptrs since `gcs_server` should own all of these in theory. That being said, as long as `gcs_server` doesn't explicitely own these objects, it's probably safest to also hold a reference to them since everyone else holding a `shared_ptr`

Reply:
I agree with that.